SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetrics.cpp

```diff
@@ -60,3 +68,156 @@ double USpatialMetrics::CalculateLoad() const
+		}
+		else
+		{
+			UE_LOG(LogSpatialMetrics, Warning, TEXT("SpatialStartRPCMetrics: Could not resolve local PlayerController entity!"));
+		}
+	}
+}
+
+void USpatialMetrics::OnStartRPCMetricsCommand()
+{
+	SpatialStartRPCMetrics();
+}
+
+void USpatialMetrics::SpatialStopRPCMetrics()
+{
+	if (!bRPCTrackingEnabled)
+	{
+		UE_LOG(LogSpatialMetrics, Log, TEXT("Haven't been recording RPC metrics"));
+		return;
+	}
+
+	// Display recorded sent RPCs.
+	const double TrackRPCInterval = FPlatformTime::Seconds() - RPCTrackingStartTime;
+	UE_LOG(LogSpatialMetrics, Log, TEXT("Recorded %d unique RPCs over the last %.3f seconds:"), RecentRPCs.Num(), TrackRPCInterval);
+
+	if (RecentRPCs.Num() > 0)
+	{
+		// NICELY log sent RPCs.
+		TArray<RPCStat> RecentRPCArray;
+		RecentRPCs.GenerateValueArray(RecentRPCArray);
+
+		// Show the most frequently called RPCs at the top.
+		RecentRPCArray.Sort([](const RPCStat& A, const RPCStat& B)
+		{
+			if (A.Type != B.Type)
+			{
+				return static_cast<int>(A.Type) < static_cast<int>(B.Type);
+			}
+			return A.Calls > B.Calls;
+		});
+
+		int MaxRPCNameLen = 0;
+		for (RPCStat& Stat : RecentRPCArray)
+		{
+			MaxRPCNameLen = FMath::Max(MaxRPCNameLen, Stat.Name.Len());
+		}
+
+		int TotalCalls = 0;
+		int TotalPayload = 0;
+
+		UE_LOG(LogSpatialMetrics, Log, TEXT("---------------------------"));
+		UE_LOG(LogSpatialMetrics, Log, TEXT("Recently sent RPCs - %s:"), NetDriver->IsServer() ? TEXT("Server") : TEXT("Client"));
+		UE_LOG(LogSpatialMetrics, Log, TEXT("RPC Type           | %s | # of calls |  Calls/sec | Total payload | Avg. payload | Payload/sec"), *FString(TEXT("RPC Name")).RightPad(MaxRPCNameLen));
+
+		FString SeparatorLine = FString::Printf(TEXT("-------------------+-%s-+------------+------------+---------------+--------------+------------"), *FString::ChrN(MaxRPCNameLen, '-'));
+
+		ESchemaComponentType PrevType = SCHEMA_Invalid;
+		for (RPCStat& Stat : RecentRPCArray)
+		{
+			FString RPCTypeField;
+			if (Stat.Type != PrevType)
+			{
+				RPCTypeField = RPCSchemaTypeToString(Stat.Type);
+				PrevType = Stat.Type;
+				UE_LOG(LogSpatialMetrics, Log, TEXT("%s"), *SeparatorLine);
+			}
+			UE_LOG(LogSpatialMetrics, Log, TEXT("%s | %s | %10d | %10.4f | %13d | %12.4f | %11.4f"), *RPCTypeField.RightPad(18), *Stat.Name.RightPad(MaxRPCNameLen), Stat.Calls, Stat.Calls / TrackRPCInterval, Stat.TotalPayload, (float)Stat.TotalPayload / Stat.Calls, Stat.TotalPayload / TrackRPCInterval);
+			TotalCalls += Stat.Calls;
+			TotalPayload += Stat.TotalPayload;
+		}
+		UE_LOG(LogSpatialMetrics, Log, TEXT("%s"), *SeparatorLine);
+		UE_LOG(LogSpatialMetrics, Log, TEXT("Total              | %s | %10d | %10.4f | %13d | %12.4f | %11.4f"), *FString::ChrN(MaxRPCNameLen, ' '), TotalCalls, TotalCalls / TrackRPCInterval, TotalPayload, (float)TotalPayload / TotalCalls, TotalPayload / TrackRPCInterval);
+
+		RecentRPCs.Empty();
+	}
+
+	bRPCTrackingEnabled = false;
```

Comment:
Reset the tracking time too?

Reply:
We reset it when starting the tracking. It's not used in between.